### PR TITLE
Add libmnl 1.0.4

### PR DESCRIPTION
**Pull request by skannan89**
Specify library name and version:  **libmnl/1.0.4**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libmnl/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+conan_basic_setup()
+
+add_executable(example example.cpp)
+target_link_libraries(example ${CONAN_LIBS})
+
+# CTest is a testing tool that can be used to test your project.
+# enable_testing()
+# add_test(NAME example
+#          WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/bin
+#          COMMAND example)
```
Comment (intelligide) [inline recipes/libmnl/all/test_package/CMakeLists.txt:14]:
```suggestion
```

File: recipes/libmnl/all/test_package/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class LibmnlTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
+
+    def build(self):
+        cmake = CMake(self)
+        # Current dir is "test_package/build/<build_id>" and CMakeLists.txt is
+        # in "test_package"
```
Comment (intelligide) [inline recipes/libmnl/all/test_package/conanfile.py:13]:
```suggestion
```

File: recipes/libmnl/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            os.chdir("bin")
+            self.run(".%sexample" % os.sep)
```
Comment (intelligide) [inline recipes/libmnl/all/test_package/conanfile.py:20]:
```suggestion
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

File: recipes/libmnl/all/test_package/example.cpp [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+#include <cstdlib>
+#include <libmnl/libmnl.h>
+
+int main() {
+    struct mnl_socket *nl = mnl_socket_open(NETLINK_NETFILTER);
+    if (nl == NULL) {
+        std::cerr << "mnl_socket_open\n";
+		return EXIT_FAILURE;
+	}
+	mnl_socket_close(nl);
```
Comment (intelligide) [inline recipes/libmnl/all/test_package/example.cpp:11]:
```suggestion
        return EXIT_FAILURE;
    }
    mnl_socket_close(nl);
```

File: recipes/libmnl/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    version = "1.0.4"
+    license = "LGPLv2.1+"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://netfilter.org/projects/libmnl/index.html"
+    description = "Minimalistic Netlink communication library"
+    topics = ("libmnl", "netlink")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+    _source_subfolder = "source_subfolder"
```
Comment (intelligide) [inline recipes/libmnl/all/conanfile.py:16]:
```suggestion

```

File: recipes/libmnl/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    _autotools = None
+
```
Comment (intelligide) [inline recipes/libmnl/all/conanfile.py:18]:
```suggestion

    @property
    def _source_subfolder(self):
        return "source_subfolder"

```

File: recipes/libmnl/all/conanfile.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
+import os, glob
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+class LibmnlConan(ConanFile):
+    name = "libmnl"
+    version = "1.0.4"
+    license = "LGPLv2.1+"
```
Comment (intelligide) [inline recipes/libmnl/all/conanfile.py:8]:
```suggestion
    license = "LGPL-2.1-or-later"
```

File: recipes/libmnl/all/conanfile.py [hunk 8]
```diff
@@ -0,0 +1,64 @@
+import os, glob
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+class LibmnlConan(ConanFile):
+    name = "libmnl"
+    version = "1.0.4"
```
Comment (intelligide) [inline recipes/libmnl/all/conanfile.py:7]:
```suggestion
```

**Review by intelligide — CHANGES_REQUESTED** [review]

File: recipes/libmnl/all/conanfile.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self)
+        config_args = [
+            "--prefix={}".format(tools.unix_path(self.package_folder)),
+        ]
```
Comment (prince-chrismc) [inline recipes/libmnl/all/conanfile.py:40]: 🤔 I would have expected this to be handled by the build helper

File: recipes/libmnl/all/conanfile.py [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        autotools.make()
+
+    def package(self):
+        autotools = self._configure_autotools()
+        autotools.install()
+        self.copy("COPYING", dst="licenses", src=self._source_subfolder)
+        la_pattern = os.path.join(self.package_folder, "lib", "**", "*.la")
+        la_files = glob.glob(la_pattern, recursive=True)
+        for next_file in la_files:
+            os.remove(next_file)
```
Comment (prince-chrismc) [inline recipes/libmnl/all/conanfile.py:60]: I am not sure about the recursion but this should work

```suggestion
        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
```

You could also remove the import for glob and set the conan mimium version to `>= 1.29.1`

File: recipes/libmnl/all/conanfile.py [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("libmnl is only supported on Linux")
```
Comment (prince-chrismc) [inline recipes/libmnl/all/conanfile.py:28]: ftp://ftp7.freebsd.org/sites/www.linuxfromscratch.org/blfs/view/systemd/basicnet/libmnl.html

```suggestion
        if not self.settings.os in ["Linux", "FreeBSD"]:
            raise ConanInvalidConfiguration("libmnl is not supported on {}".format(self.settings.os))
```

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libmnl/all/conanfile.py (hunk 11 above)
Comment (skannan89) [reply recipes/libmnl/all/conanfile.py:30]: Addressed rest of your feedback. Their git repo says it's linux only. https://git.netfilter.org/libmnl/tree/configure.ac#n21

Comment (prince-chrismc) [reply recipes/libmnl/all/conanfile.py:30]: Gross! Okay that's fine.

**Review by madebr — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**